### ISSUE TITLE
`git lfs fetch` on a freshly cloned repo gives `error: git-lfs died of signal 13`

### ISSUE THREAD

**Issue by edzis** [github/git-lfs#591]
Freshly cloned repo, either via https or ssh
with some files removed before the master

**Comment by technoweenie** [comment]
I think this is a dupe of https://github.com/github/git-lfs/issues/588. If that matches your issue, I'll close this.

**Comment by edzis** [comment]
Seems related, but have not found any mentions of signal 13 (broken pipe) in relation to `git lfs fetch`

**Comment by technoweenie** [comment]
Can you post a gist with the details?

1. `git lfs env`
2. `git lfs logs last` - This will dump the stacktrace of the most recent exception. If that's not the same exception, use `git lfs logs` to list errors that you've seen in that repo, and `git lfs logs show <filename>` to view it.
3. `GIT_TRACE=1 git lfs fetch` will print out internal info while running the fetch command. Could help pinpoint where the broken pipe is coming from.

**Comment by arikon** [comment]
@edzis Could you provide more info on this so it could be fixed?

**Comment by cjbottaro** [comment]
```$ git lfs checkout
Git LFS: (1 of 118 files) 229.00 KB / 19.76 MB                                                                                                                                                                                        error: git-lfs died of signal 13

$ git lfs logs last
No logs to show

$ git lfs env
git-lfs/1.1.1 (GitHub; linux amd64; go 1.5.3; git 7de0397)
git version 1.9.1

Endpoint=https://github.com/AcademicWorks/user_app.git/info/lfs (auth=none)
  SSH=git@github.com:AcademicWorks/user_app.git
LocalWorkingDir=/home/rails/user_app/releases/12812af940e376ad75587395eaaf0d316e551860
LocalGitDir=/home/rails/user_app/current/.git
LocalGitStorageDir=/home/rails/user_app/current/.git
LocalMediaDir=/home/rails/user_app/current/.git/lfs/objects
TempDir=/home/rails/user_app/current/.git/lfs/tmp
ConcurrentTransfers=3
BatchTransfer=true
git config filter.lfs.smudge = ""
git config filter.lfs.clean = ""
```

**Comment by technoweenie** [comment]
How about `GIT_TRACE=1 git lfs checkout`?

**Comment by cjbottaro** [comment]
```
$ GIT_TRACE=1 git lfs checkout
trace: exec: 'git-lfs' 'checkout'
trace: run_command: 'git-lfs' 'checkout'
trace git-lfs: run_command: 'git' rev-parse HEAD --symbolic-full-name HEAD
trace git-lfs: run_command: git ls-tree -r -l --full-tree 12812af940e376ad75587395eaaf0d316e551860
trace git-lfs: run_command: git cat-file --batch
Git LFS: (1 of 118 files) 229.00 KB / 19.76 MB                                                                                                                                                                                        error: git-lfs died of signal 13
```

**Comment by technoweenie** [comment]
Weird, it'd sure be nice if it dumped any useful error message at all :)

How about `git lfs ls-files` vs `tree .git/lfs/objects`? I want to see the state of your local repository so I can potentially recreate it. Specifically, I'm checking to see if you have all of the files downloaded to `.git/lfs/objects`. The fetch/checkout/pull commands are similar to their git versions:

* fetch - downloads the LFS objects, but does not replace LFS pointers
* checkout - replaces LFS pointers with files in .git/lfs/objects, does not download them though.
* pull - fetch + checkout

Also, if this is a private repo, send details to https://github.com/contact if you want to take this to a private channel. Or ping me in a private msg on https://gitter.im/github/git-lfs if you want.

**Comment by autious** [comment]
I have this issue too on 1.1.0. Re-running git lfs multiple times sometimes helps (it gets a little further every time).

**Comment by technoweenie** [comment]
How's this on v1.3.x? There have been some good bug fixes since v1.1 when a repo has a lot of refs or objects. Re-open if you have good `GIT_TRACE=1` output.

**Comment by kbespalyi** [comment]
I found from another post - works!
<
There are multiple versions of git-lfs installed on your computer. Try:
$ rm "$(which git-lfs)"
$ brew link git-lfs --overwrite
>
